### ISSUE TITLE
AREPL Encoder cp1252 to utf-8

### ISSUE THREAD

**Issue by NigelBloom** [Almenon/AREPL-vscode#329]
I'm new to AREPL so if I'm just overlooking a simple feature that already exists or it's user error feel free to call me out.

When I run the code:
`import sys
print(sys.stdout.encoding)`

AREPL prints: cp1252
VSCode in the terminal prints: utf-8
Also VSCode encoder is set to UTF-8

![Encoder](https://user-images.githubusercontent.com/64054634/79820829-34d72900-835b-11ea-806c-3df1a02d0f0b.PNG)

This difference can be quite inconvenient when trying to pull data from a web page and AREPL returns "UnicodeEncodeError: 'charmap' codec can't encode characters in position 0-4: character maps to <undefined>",

**Comment by Almenon** [comment]
That's really weird, I get the same thing. Not sure why. 🤔 

I think I can fix it by running `sys.stdout.reconfigure(encoding=sys.getdefaultencoding())` in the backend. Or maybe I can set the stdout encoding when spawning the python process.

As a temporary workaround you can add`"import sys;sys.stdout.reconfigure(encoding=sys.getdefaultencoding())"` to the AREPL.defaultImports setting so that line of code will always be be ran.

Do you have an example of code where you get that UnicodeEncodeError error? I don't think my fix will fix the UnicodeEncodeError unfortunately. I tested it and if I try to print the eggplant emoji I get `UnicodeEncodeError: 'utf-8' codec can't encode character '\udc8d' in position 56: surrogates not allowed`. It's a bug that's [been around for a while](https://github.com/Almenon/AREPL-vscode/issues/149) but I have had no idea how to fix it. Now I'm thinking maybe I just need to set the stdin encoding to utf8.

**Comment by NigelBloom** [comment]
Got it, let me know what you think.

Here is an example of code I'm running:
`import requests
response = requests.get('https://randomuser.me/api/?results=10')
data = response.json()
for user in data['results']:
    print(user['name']['first'])`

You can see that AREPL printed 6 names before running into the error and the Terminal did all 10 names.
![UnicodeEncodeError Code](https://user-images.githubusercontent.com/64054634/79929284-4a0f8e80-8413-11ea-9932-95c9b2ba9dc7.PNG)



I'm currently using different variations of `.encode("utf-8")` just to keep AREPL running with no errors, but won't work as a final solution.
![A work around](https://user-images.githubusercontent.com/64054634/79929365-7fb47780-8413-11ea-98df-2915c89bd8f5.PNG)

**Comment by Almenon** [comment]
I solved it! :D 

Take that unicode errors!

I looked into it and found that the python-shell library I use in the backend already uses utf8 encoding, so it wasn't node's fault. After reading through the python docs I found out that in python pipes use the system locale encoding (cp1252 in case of windows). The console device uses utf8 which is why python works correctly in the terminal. You can create a .env file in the root folder of your workspace with `PYTHONIOENCODING=utf8` (see explanation of var in [python docs](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONIOENCODING)) and emojis and other unicode goodness will work in AREPL :)

Of course having to create a .env file in every project would be annoying - I'll release a update to arepl that sets PYTHONIOENCODING=utf8 automatically.

---

Random sidenote - for interacting with API's I suggest putting the code below a comment `#$end` and using control-enter to evaluate your blocks of code. This way the API doesn't called every time you stop typing, which might be more desirable behavior than real-time because API's can take a while to respond. Network latency and all that. Also less requests for the API to have to deal with.

**Comment by GitToh** [comment]
![messageImage_1587818244214](https://user-images.githubusercontent.com/63533552/80281505-cec11c00-8735-11ea-8d8e-20e7a9a8f22c.jpg)
-- TRY
`<html lang="UTF-8"`

-- OR
```
<head>
  <meta charset="UTF-8">
</head>
```

**Comment by Almenon** [comment]
@GitToh why is that? Have you found a bug with the landing page? Emojis work fine in there for me despite the lang="en". Although maybe that is just because of the region I'm in 🤔 

Also I don't think utf-8 is a possible value for lang. Lang is to specify the language, but utf-8 is a character encoding.

**Comment by GitToh** [comment]
sorry for commented.
I think this is better 
```
<head>
  <meta charset="UTF-8">
</head>
```

**Comment by mthekid** [comment]
hmm i don't think so it is really helpful answer...

I am Korean and i used korean characters in comment.

but arepl didn't works when korean character in code.

I found some trouble issues in Encoding tech in Korea.

There is some trouble when using EUC-KR encoding tech and my OS using cp949...

and i change my encoding setting in windows system.

control -> World&National -> Administration Option -> change the system Local -> Check box : Beta :~~~~~~ Unicode UTF-8 using.

and after rebooting my computer.

there is no problem using arepl in vs code..

My English writing skills are not good....

but I hope it will help some people..

**Comment by Almenon** [comment]
I just released AREPL 1.0.24!
AREPL should now work with unicode on Windows. No .env required. Not sure about linux/mac.
Sorry about the delay, I had a absolutely horrendous time dealing with a [stdout issue](https://github.com/Almenon/AREPL-vscode/pull/332).

@mthekid @GitToh @NigelBloom can you test it and confirm if it works now please?
